src/main/kotlin/com/terraformation/backend/accelerator/api/ProjectModulesController.kt

```diff
@@ -0,0 +1,97 @@
+package com.terraformation.backend.accelerator.api
+
+import com.terraformation.backend.accelerator.db.ModuleStore
+import com.terraformation.backend.accelerator.model.EventModel
+import com.terraformation.backend.accelerator.model.ModuleModel
+import com.terraformation.backend.api.AcceleratorEndpoint
+import com.terraformation.backend.api.ApiResponse200
+import com.terraformation.backend.api.ApiResponse404
+import com.terraformation.backend.db.accelerator.EventId
+import com.terraformation.backend.db.accelerator.EventType
+import com.terraformation.backend.db.accelerator.ModuleId
+import com.terraformation.backend.db.default_schema.ProjectId
+import io.swagger.v3.oas.annotations.Operation
+import java.net.URI
+import java.time.Instant
+import java.time.LocalDate
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+
+@AcceleratorEndpoint
+@RequestMapping("/api/v1/modules")
+@RestController
+class ProjectModulesController(
+    private val moduleStore: ModuleStore,
+) {
+  @ApiResponse200
+  @ApiResponse404
+  @GetMapping("/projects/{projectId}")
+  @Operation(summary = "Gets modules for a project.")
+  fun listModules(
+      @PathVariable("projectId") projectId: ProjectId,
+  ): GetProjectModulesResponsePayload {
+    val models = moduleStore.fetchModulesForProject(projectId)
+    return GetProjectModulesResponsePayload(models.map { ProjectModule(it) })
+  }
+}
+
+data class ProjectModuleEventSession(
+    val id: EventId,
+    val startTime: Instant?,
+    val endTime: Instant?,
+    val meetingURL: URI?,
+    val recordingURL: URI?,
+    val slidesURL: URI?,
+) {
+  constructor(
+      model: EventModel
+  ) : this(
+      id = model.id,
+      startTime = model.startTime,
+      endTime = model.endTime,
+      meetingURL = model.meetingURL,
+      recordingURL = model.recordingURL,
+      slidesURL = model.slidesURL,
+  )
+}
+
+data class ProjectModuleEvent(
+    val eventDescription: String,
+    val sessions: List<ProjectModuleEventSession>,
+)
+
+data class ProjectModule(
+    val id: ModuleId,
+    val name: String,
```

Comment:
Do we also need a "module title" or something like that? We need "Module 1" and "Kick-off" in the frontend.

Reply:
I was thinking that the Module X part comes from the ordering of modules. "Kick-off" is the name in this case. 

The query for project modules currently sort modules by startDate. And the table is constraint so that there is no overlapping dates. 

We may want to add an isActive boolean that comes from the backend.

Reply:
And any given module can be in multiple cohorts, in any order. So we can't hard-wire the Module X title either.

Reply:
We have a table modules, and cohort_modules that associate modules with cohorts

modules:
- name
- description
- html text 
etc. most of the contents

cohort_modules:
- module_id
- cohort_id
- start_date
- end_date (constraint date collisions and end_date > start_date)

Reply:
And adding to before, the isActive boolean will be there to make sure we are respecting server clock and dates as opposed to browser clock which can be set to anything.

Reply:
Oh I thought the content of the modules could change from cohort to cohort

Reply:
We are going with every time a module changes it becomes a brand new module.

Reply:
We discussed with product and they didn't want accelerator admins to deal with the complexity of content management. So if module has any variations, it is a brand new module.

Reply:
Sounds good, thanks for the info